### PR TITLE
ARUHA-2124 support for filtering lists

### PR DESCRIPTION
**Pull request by lmontrieux**
ARUHA-2124: Add filter capability to plugin API

## Description
A new method, `List<Object> filter(List<Object> list)` to filter a list according to what the subject should be allowed to receive

## Review
- [ ] Tests
- [ ] Documentation
- [ ] CHANGELOG

### CODE REVIEW

File: src/main/java/org/zalando/nakadi/plugin/api/authz/AuthorizationService.java [hunk 1]
```diff
@@ -32,4 +34,16 @@
...
+     * Filters a list based on authorization rules.
+     *
+     * For example, if a user is only able to see the event types for which he has some permissions, this method
+     * will take the list of all event types, and return only those that the caller can see.
+     *
+     * @param input the list to filter
+     * @return a filtered list
+     * @throws PluginException if an error occurred during execution
+     */
+    List<Object> filter(List<Object> input) throws PluginException;
```
Comment (rcillo) [inline src/main/java/org/zalando/nakadi/plugin/api/authz/AuthorizationService.java]: Should the input list be a list of `Object` or a list of `Resources` from which we can derive some information about who is allowed to access it?

Comment (lmontrieux) [reply src/main/java/org/zalando/nakadi/plugin/api/authz/AuthorizationService.java]: Updated in in 8396d33 for a list of resources.

**Comment by rcillo** [comment]
👍

**Review by rcillo — APPROVED** [review]

**Comment by lmontrieux** [comment]
👍

**Comment by ferbncode** [comment]
:+1:

**Pull request merged**